WordPress/src/test/java/org/wordpress/android/ui/mysite/MySiteViewModelTest.kt

```diff
@@ -2163,6 +2163,82 @@ class MySiteViewModelTest : BaseUnitTest() {
+        initSelectedSite(isMySiteDashboardTabsFeatureFlagEnabled = true, isMySiteTabsBuildConfigEnabled = true)
+
+        requireNotNull(quickLinkRibbonPostsClickAction).invoke()
+
+        assertThat(trackWithTabSource.last().stat).isEqualTo(Stat.QUICK_LINK_RIBBON_POSTS_TAPPED)
+    }
+
+    @Test
+    fun `given tabs are enabled, when quick link ribbon stats tapped, then track with tab source is requested`() {
+        initSelectedSite(isMySiteDashboardTabsFeatureFlagEnabled = true, isMySiteTabsBuildConfigEnabled = true)
+
+        requireNotNull(quickLinkRibbonStatsClickAction).invoke()
+
+        assertThat(trackWithTabSource.last().stat).isEqualTo(Stat.QUICK_LINK_RIBBON_STATS_TAPPED)
+    }
+
+    @Test
+    fun `given tabs are enabled, when quick link ribbon media tapped, then track with tab source is requested`() {
+        initSelectedSite(isMySiteDashboardTabsFeatureFlagEnabled = true, isMySiteTabsBuildConfigEnabled = true)
+
+        requireNotNull(quickLinkRibbonMediaClickAction).invoke()
+
+        assertThat(trackWithTabSource.last().stat).isEqualTo(Stat.QUICK_LINK_RIBBON_MEDIA_TAPPED)
+    }
+
+    @Test
+    fun `given tabs are disabled, when quick link ribbon stats tapped, then track with tab source is not requested`() {
+        initSelectedSite(isMySiteDashboardTabsFeatureFlagEnabled = false, isMySiteTabsBuildConfigEnabled = false)
+
+        requireNotNull(quickLinkRibbonStatsClickAction).invoke()
+
+        assertThat(trackWithTabSource).isEmpty()
+        assertThat(analyticsTrackerWrapper.track(Stat.QUICK_LINK_RIBBON_STATS_TAPPED))
+    }
+
+    @Test
+    fun `given tabs are disabled, when quick link ribbon pages tapped, then track with tab source is not requested`() {
+        initSelectedSite(isMySiteDashboardTabsFeatureFlagEnabled = false, isMySiteTabsBuildConfigEnabled = false)
+
+        requireNotNull(quickLinkRibbonPagesClickAction).invoke()
+
+        assertThat(trackWithTabSource).isEmpty()
+        assertThat(analyticsTrackerWrapper.track(Stat.QUICK_LINK_RIBBON_PAGES_TAPPED))
+    }
+
+    @Test
+    fun `given tabs are disabled, when quick link ribbon posts tapped, then track with tab source is not requested`() {
+        initSelectedSite(isMySiteDashboardTabsFeatureFlagEnabled = false, isMySiteTabsBuildConfigEnabled = false)
+
+        requireNotNull(quickLinkRibbonPostsClickAction).invoke()
+
+        assertThat(trackWithTabSource).isEmpty()
+        assertThat(analyticsTrackerWrapper.track(Stat.QUICK_LINK_RIBBON_POSTS_TAPPED))
+    }
+
+    @Test
+    fun `given tabs are disabled, when quick link ribbon media tapped, then track with tab source is not requested`() {
+        initSelectedSite(isMySiteDashboardTabsFeatureFlagEnabled = false, isMySiteTabsBuildConfigEnabled = false)
+
+        requireNotNull(quickLinkRibbonMediaClickAction).invoke()
+
+        assertThat(trackWithTabSource).isEmpty()
+        assertThat(analyticsTrackerWrapper.track(Stat.QUICK_LINK_RIBBON_MEDIA_TAPPED))
+    }
```

Comment:
❓ I'm wondering if we need `given tabs are disabled` test cases. Original quick actions are shown even when tabs are disabled. But the `quick link ribbon` is available only on `Home`/ `Dashboard` tab. We will never be able to execute the`when clause` in the app as quick link ribbon is not built when tabs are disabled. 

We might want to add test cases in the `org.wordpress.android.ui.mysite.cards.CardsBuilder` whether a quick link ribbon  is built. 

Wdyt?

Reply:
> I'm wondering if we need given tabs are disabled test cases. Original quick actions are shown even when tabs are disabled. But the quick link ribbon is available only on Home/ Dashboard tab. We will never be able to execute thewhen clause in the app as quick link ribbon is not built when tabs are disabled.


Yup, you are right. 💯 . What was I thinking 🤔 . I have removed the disabled test cases in the commit. [65907ef](https://github.com/wordpress-mobile/WordPress-Android/pull/16284/commits/65907efbcd61a0d12edfb97b1d1d2878cb6a07eb)


>We might want to add test cases in the org.wordpress.android.ui.mysite.cards.CardsBuilder whether a quick link ribbon is built.

I have added the test case for the condition `if tabs are disabled, then quick link should not be built` in `CardsBuilderTest.kt`. Please find the details below. 

Also I am not sure how those tabs disabled tests passed in `MyViewModelTest.kt`. I believe the initialisation for those tests was wrong, I didn't go deeper on how those succeeded as those tests were not needed anyways. 

<img width="1050" alt="Screenshot 2022-04-08 at 6 33 35 PM" src="https://user-images.githubusercontent.com/17463767/162441208-b694eb8f-8e7d-48b1-8e6c-f36cfc76a444.png">

Reply:
I can see card builder tests for the ribbon, so we're good. Thank you for removing redundant test cases. 😄